### DISCUSSION TITLE
24.11

### DISCUSSION THREAD

**Discussion by gaving** [nix-community/NixOS-WSL#610] [Q&A] [answered]
Hi! 👋 

Thanks for the awesome project, it's a lifesaver for me at work!

Is 24.11 support is on the roadmap any time soon? Is it as simple as repointing my `nixpkgs` URLs to the `24.11` branch and it'll continue to work with the `24.05` os?

Figured I'd ask before interfering with my work env.

**Comment by sedlund** [comment, marked as answer]
yes, you can point to 24.11.

>  it'll continue to work with the 24.05 os?

if you have one configuration the whole system will be upgraded to whatever you point nixpkgs at (most probably are running unstable for reference).  if you have separate home-manager profile you should not mix channels by having a different OS and home profile version (generally...)

**Reply by gaving** [reply]
Cheers for the response - yeah I'm on 24.05 for NixOS WSL with a 24.05 home manager profile, from my `flake.nix`:-

```nix
  inputs = {
    _1password-shell-plugins.url = "github:1Password/shell-plugins";
    home-manager.inputs.nixpkgs.follows = "nixpkgs";
    home-manager.url = "github:nix-community/home-manager/release-24.05";
    impermanence.url = "github:nix-community/impermanence";
    neovim-nightly-overlay.url = "github:nix-community/neovim-nightly-overlay";
    nix-darwin.inputs.nixpkgs.follows = "nixpkgs-darwin";
    nix-darwin.url = "github:LnL7/nix-darwin/master";
    nixos-wsl.inputs.nixpkgs.follows = "nixpkgs";
    nixos-wsl.url = "github:nix-community/NixOS-WSL";
    nixpkgs-darwin.url = "github:NixOS/nixpkgs/nixpkgs-24.05-darwin";
    nixpkgs-unstable.url = "github:nixos/nixpkgs/nixos-unstable";
    nixpkgs.url = "github:nixos/nixpkgs/nixos-24.05";
    nixvim.inputs.nixpkgs.follows = "nixpkgs";
    nixvim.url = "github:nix-community/nixvim/nixos-24.05";
    sops-nix.url = "github:Mic92/sops-nix";
  };
```

So presumably should hold off updating home-manager in this case?

**Reply by sedlund** [reply]
hm made a 24.11 branch a few weeks ago 
https://github.com/nix-community/home-manager?tab=readme-ov-file#releases

If they hadn't, one could just target their master branch until it came out as they are very similar that close to release.  Ymmv ;)

**Reply by gaving** [reply]
Thanks! I'll see how I'll get on.

I'll look to do a fresh nixos-wsl install if/when a 24.11 release is cut anyway.

(I'm in an environment with an average link speed of 50k/s hence all this hesitation for just trying it!)

**Reply by sedlund** [reply]
ah well if something gives you a problem you can point back to 24.05 and activate it as long as you don't nix gc everything right away.

also have this page handy  https://nix-community.github.io/NixOS-WSL/troubleshooting/recovery-shell.html

but should be fine :) 👍

**Reply by gaving** [reply]
Yeah that's true.

Appreciate the help, thanks again!
